### PR TITLE
StreamElement.upsert and deprecate StreamElement.update

### PR DESCRIPTION
**Pull request by LesTR**

### CODE REVIEW

**Comment by je-ik** [comment]
LGTM

**Comment by je-ik** [comment]
I'll experiment with this a bit to see if I can get there the nicer upsert/delete APIs to construct StreamElements from ModelGenerator. I'll merge it then.

**Comment by LesTR** [comment]
> I'll experiment with this a bit to see if I can get there the nicer upsert/delete APIs to construct StreamElements from ModelGenerator. I'll merge it then.

I agree but we can do that in another PR. Keep it simple.

**Comment by je-ik** [comment]
@LesTR I think we can easily review this here. Can you have a look a [this commit](https://github.com/datadrivencz/proxima-platform/pull/269/commits/c512914c145978ca325798f199b9c20575045c11)

File: maven/src/main/resources/java-source.ftlh [hunk 1]
```diff
@@ -167,17 +170,19 @@ public class ${java_classname} implements Serializable {
   <#list entity.attributes as attribute>
     <#if attribute.wildcard>
     @SuppressWarnings("unchecked")
-    private final AttributeDescriptor<${attribute.type}> ${attribute.name}Descriptor =
-      (AttributeDescriptor) descriptor.getAttribute("${attribute.name}.*");
+    private final Wildcard<${attribute.type}> ${attribute.name}Descriptor =
+      EntityAwareAttributeDescriptor.wildcard(descriptor, (AttributeDescriptor) descriptor.getAttribute("${attribute.name}.*"));
```
Comment (LesTR) [inline maven/src/main/resources/java-source.ftlh]: We know the types so casting to `(AttributeDescriptor)` can be imho removed and also `@SuppressWarnings("unchecked")` right?

File: core/src/main/java/cz/o2/proxima/repository/EntityAwareAttributeDescriptor.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+          entity,
+          wrapped,
+          uuid,
+          key,
+          wrapped.getName(),
+          stamp,
+          wrapped.getValueSerializer().serialize(value));
+    }
+
+    public StreamElement upsert(String uuid, String key, Instant stamp, T value) {
```
Comment (LesTR) [inline core/src/main/java/cz/o2/proxima/repository/EntityAwareAttributeDescriptor.java]: Big 👍 !
minor: Can you please add javadocs?

File: maven/src/main/resources/java-source.ftlh (hunk 1 above)
Comment (je-ik) [reply maven/src/main/resources/java-source.ftlh]: :+1:

File: core/src/main/java/cz/o2/proxima/repository/EntityAwareAttributeDescriptor.java (hunk 2 above)
Comment (je-ik) [reply core/src/main/java/cz/o2/proxima/repository/EntityAwareAttributeDescriptor.java]: Added.

**Pull request merged**